react-native-image-picker/react-native-image-picker
# Update handling of Android runtime permission - keep showing system's permission dialog whenever possible (fixes #739)

## Motivation 

See #739. Currently, the library redirects users to app's setting view after user deny the permission once, which leads to suboptimal user experience. 
This PR changes behaviour as described in test plan 1 below.

## Test Plan

#### Test plan 1
1. install Example app on Android 6.0+ (screenshot [1])
2. Click select photo, select option
3. Dialog to ask permission is shown (screenshot [2])
4. Deny permission request
5. Click select photo, select option again
6. Dialog to explain why permission is needed shown  (screenshot [3]) (Note behaviour from 1 to 6 are same as before this PR)
7. Now, click try again. The permission dialog is shown again, now with 'never ask again' checkbox (screenshot [4])
8. Deny permission. You'll see (screenshot [3]) again and now clicking 'try again' leads to app settings.
9. If you click 'select photo' again, you'll see (screenshot [3]) again and clicking 'try again' leads to app settings.

#### Test plan 2 (nothing changed, just make sure there is no regression)
1. install Example app on Android 6.0+ (screenshot [1])
2. Click select photo, select option
3. Dialog to ask permission is shown (screenshot [2])
4. Accept permission requests and you can use image picker

[1]
![screen shot 2017-11-24 at 12 55 12 pm](https://user-images.githubusercontent.com/980077/33224716-45c9db66-d121-11e7-9fe0-14ba0ac2661f.png)

[2]
![screen shot 2017-11-24 at 2 11 33 pm](https://user-images.githubusercontent.com/980077/33224737-7a69b030-d121-11e7-8a86-23117601ec48.png)

[3]
![screen shot 2017-11-24 at 12 55 37 pm](https://user-images.githubusercontent.com/980077/33224715-45b5a89e-d121-11e7-95a7-e3cacf0bddb9.png)

[4]
![screen shot 2017-11-24 at 12 55 47 pm](https://user-images.githubusercontent.com/980077/33224714-459d26de-d121-11e7-8882-ee51dc20e9ec.png)
